### ISSUE TITLE
WebService API - create new order - id_carrier and total_shipping ignored

### ISSUE THREAD

**Issue by rbento1096** [PrestaShop/PrestaShop#32622]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I am developing an e-commerce application whose products are synced with a prestashop database. When a purchase is completed, I create an order through the Webservice API using the following object (which is converted to XML). The order is successfully created but when I GET the order from the API the `id_carrier` is equal to `0` and the `total_shipping` is equal to `5` (which I guess is the default carrier's cost).

Is there any reason the API is overriding my provided values for `id_carrier` and `total_shipping`? The same seems to be the case also when I pass values for the `total_discounts` attribute.

Besides, I have to pass the same value (`totalAmountWithOriginalPrices`) to both the `total_paid` and `total_products_wt` attributes, otehrwise I get a `PaymentError` on my Prestashop dashboard.

``` javascript
const orderObj = {
      prestashop: {
        order: {
          id_address_delivery: deliveryAddressId,
          id_address_invoice: billingAddressId,
          id_cart: cartId,
          id_currency: 2,
          id_lang: 2,
          id_customer: customerId,
          id_carrier: carrierId,
          module: 'paypal',
          payment: 'payment-app',
          total_paid: totalAmountWithOriginalPrices,
          total_paid_real: totalAmountWithOriginalPrices,
          total_products: totalAmountWithOriginalPrices,
          total_products_wt: totalAmountWithOriginalPrices,
          total_shipping: purchase.shippingOption?.amount || 0,
          total_shipping_tax_incl: purchase.shippingOption?.amount || 0, 
          total_shipping_tax_excl: purchase.shippingOption?.amount || 0, 
          conversion_rate: 1
        }
      }
    };
```

### Expected behavior

I am expecting the API to read my provided values and set them in the order object or at least calculate the correct values based on its input

### Steps to reproduce

Currently I can't release the app so you can try to reproduce the steps since I'm blocked on this step for the client.

But a POST request to the webservice API should suffice. (note: a customer, address and cart must be created beforehand for it to work)

### PrestaShop version(s) where the bug happened

1.7.7.1

**Comment by hibatallahAouadni** [comment]
Hello @rbento1096 

We are aware of this issue, it is already in our debug backlog. This issue is a duplicate of #19906. 
To be informed when it's fixed, please subscribe to the issue mentioned above.

Thanks!

**Comment by hibatallahAouadni** [comment]
Duplicate of #19906

**Comment by rbento1096** [comment]
Hello @hibatallahAouadni I believe I have found the issue. I didn't have permissions to edit other tables such as `order_carriers` (which I assume is written to during the `POST order` operation). After granting those permissions it still didn't work yesterday but I tried again today and the correct `id_carrier` is now being passed (maybe a cache issue?).

**Comment by hibatallahAouadni** [comment]
Glad to hear :tada:
